Weather Function:

CodingRoy opened the pull request:
Mention the bot and say: Weather 'city' 'state/country'
Example:

> @[name-of-bot] Weather Amsterdam NL

CodingRoy committed:
Weather Function:
Mention the bot and say: Weather 'city' 'state/Country'

CodingRoy committed:
Update index.js

CodingRoy committed:
Weather for just 1 day

CodingRoy committed:
Merge branch 'function-Weather' of https://github.com/CodingRoy/bot into function-Weather

Conflicts:
	src/index.js

CodingRoy committed:
Improved tab's

CodingRoy committed:
typo

CodingRoy commented:
And don't forget to add the Wunderground API key of course ;) xD

stephan281094 commented on src/index.js:
```diff
@@ -1,35 +1,70 @@
 var Botkit = require('botkit')
 var controller = Botkit.slackbot()
+var http = require('http');
+var weatherkey = 'Your wunderground key'; //wunderground api key
```
Please use the configuration variable from `.npmrc`:

``` js
process.env.WUNDERGROUND_API_KEY || process.env.npm_config_WUNDERGROUND_API_KEY
```

stephan281094 commented on src/index.js:
```diff
@@ -1,35 +1,70 @@
 var Botkit = require('botkit')
 var controller = Botkit.slackbot()
+var http = require('http');
```
Please respect the standardjs code styling. Check linting errors by running `npm run lint`.

stephan281094 commented on src/index.js:
```diff
@@ -1,35 +1,70 @@
+      ' condition: ' + days[1].conditions);
+      if(days[1].icon == 'clear'){
+        bot.reply(message, ':sunny:');
+      } else if(days[1].icon == 'partlycloudy' || days[1].icon == 'partlysunny'){
```
For readability, use a switch statement instead.

stephan281094 commented on src/index.js:
```diff
@@ -1,35 +1,70 @@
+
+controller.hears("weather (.*) (.*)", ["mention", "direct_mention", "direct_message"], function(bot,message){
+  var city = message.match[1];
+  var state = message.match[2]; // state can also be a country but use states for the US!
```
Make the state argument optional by setting a default argument based on a configuration setting. E.g. `WUNDERGROUND_DEFAULT_STATE` inside `.npmrc`. Make sure to explain this in the readme.

stephan281094 commented on src/index.js:
```diff
@@ -1,35 +1,70 @@
+  var city = message.match[1];
+  var state = message.match[2]; // state can also be a country but use states for the US!
+  var url = '/api/' + weatherkey + '/forecast/q/state/city.json'
+  url = url.replace('state', state);
```
Instead of replacing both `state` and `city`, use string concatenation.

CodingRoy committed:
Weather API key in .npmrc file

The Weather APi key shoud now be stated in the .npmrc file like :
>WEATHER_API-KEY = yourwundergroundapikey

CodingRoy committed:
Fixed all errors
By running npm run lint

CodingRoy committed:
All improvements done

needs to be looked at the state/country.
Can't find a way to make it optional.

CodingRoy committed:
Edited readme

CodingRoy commented on src/index.js:
```diff
@@ -1,35 +1,81 @@
+
+controller.hears('weather (.*)', ['mention', 'direct_mention', 'direct_message'], function (bot, message) {
+  var city = message.match[1] + '.json'
+  var state = process.env.WEATHER_DEFAULT_STATE || process.env.npm_config_WEATHER_DEFAULT_STATE  // state can also be a country but use states for the US!
```
The state is still not optional, I can't find a way to make it optional atm.
Looks like something you can improve ;)

CodingRoy committed:
typo
